lib/common/dsr.c

```diff
@@ -29,27 +29,36 @@
 h2o_iovec_t h2o_dsr_serialize_req(h2o_dsr_req_t *req)
 {
     h2o_iovec_t output = h2o_iovec_init(
-        h2o_mem_alloc(sizeof("quic=4294967295, cipher=65535, address=\"[0000:1111:2222:3333:4444:5555:6666:7777]:65535\"")), 0);
-
-    output.len +=
-        sprintf(output.base + output.len, "quic=%" PRIu32 ", cipher=%" PRIu16 ", address=\"", req->quic_version, req->cipher);
-    switch (req->address.sa.sa_family) {
-    case AF_INET:
-        inet_ntop(AF_INET, &req->address.sin.sin_addr, output.base + output.len, INET_ADDRSTRLEN);
-        output.len += strlen(output.base + output.len);
-        output.len += sprintf(output.base + output.len, ":%" PRIu16, ntohs(req->address.sin.sin_port));
-        break;
-    case AF_INET6:
-        output.base[output.len++] = '[';
-        inet_ntop(AF_INET6, &req->address.sin6.sin6_addr, output.base + output.len, INET6_ADDRSTRLEN);
-        output.len += strlen(output.base + output.len);
-        output.len += sprintf(output.base + output.len, "]:%" PRIu16, ntohs(req->address.sin.sin_port));
-        break;
-    default:
-        h2o_fatal("unexpected address family");
-        break;
+        h2o_mem_alloc(sizeof(
+            "http-version=4294967295, quic=4294967295, cipher=65535, address=\"[0000:1111:2222:3333:4444:5555:6666:7777]:65535\"")),
+        0);
+    assert(req->http_version >= 0);
+
+    output.len += sprintf(output.base + output.len, "http-version=%d", req->http_version);
```

Comment:
Maybe omit "-version" for consistency?

QUIC version was and continues to be represented by "quic". Do we want to change that, or change "http-version" to "http".